yarn/check.py

```diff
@@ -217,11 +222,52 @@ class YarnCheck(AgentCheck):
         'queue',
         'user'
     ]
+    
+    # Our Authentication Method. Defaults to None
+    requests_authentication = None
+    
+    # On initialize, handle anything related to Kerberos authentication
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(YarnCheck, self).__init__(name, init_config, agentConfig, instances)
+        kerberos_auth = self.init_config.get('kerberos_auth', False)
+        basic_auth = self.init_config.get('basic_auth', False)
+        digest_auth = self.init_config.get('digest_auth', False)
+        
+        # Kerberos Authentication
+        if kerberos_auth:
+            kerberos_user = self.init_config.get('kerberos_user', "")
+            kerberos_realm = self.init_config.get('kerberos_realm', "")
+            kerberos_password = self.init_config.get('kerberos_password', "")
+            # Run kinit via subprocess
```

Comment:
Could you please leave a small comment here stating that this is okay because we're doing this in the `init`? 

Context: Because Agent 6 runs python checks concurrently we don't allow `Popen` to be called directly from checks (the subprocess module isn't 100% thread-safe, and in the past we've dealt with this leading to rogue processes being created when we run python checks concurrently in different threads), but since `init` is called by the python check loader there shouldn't be concurrency issues. If this becomes a problem in a future we can use the `get_subprocess_output` function which `utils` provides, but to do that we'd either have to modify that to support piping from stdin (so we can pass the password) or we'd have to store these credentials in a keytab and pass that to `kinit` directly.